tests/integration/algod_test.py

```diff
@@ -0,0 +1,76 @@
+from unittest import mock
+
+import pytest
+
+from pyteal.errors import AlgodClientError
+from pyteal.util import algod_with_assertion
+
+
+def test_algod_with_assertion_copacetic():
+    """
+    In C.I. integration tests we expect the happy path for the Algod client
+    """
+    client = algod_with_assertion()
+
+    assert client
+
+    reclient = algod_with_assertion(client)
+    assert client is reclient
+
+
+def test_algod_errors():
```

Comment:
are these taken from someplace?

Reply:
These are mostly new tests because I'm exposing `algod_with_assertion()` into the non-test code path and introducing the new exception `AlgodClientError`. Exposing algod this way in PyTeal is a bit controversial as previously discussed with @algochoi .